kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt

```diff
@@ -483,7 +483,10 @@ internal fun TargetProperty.generator(
 private fun List<AnnotationSpec>?.qualifiers(elements: Elements): Set<AnnotationSpec> {
   if (this == null) return setOf()
   return filterTo(mutableSetOf()) {
-    elements.getTypeElement(it.typeName.toString()).getAnnotation(JSON_QUALIFIER) != null
+    val typeElement = requireNotNull(elements.getTypeElement(it.typeName.rawType().canonicalName)) {
```

Comment:
`checkNotNull` is more appropriate here as this is not an argument

```suggestion
    val typeElement = checkNotNull(elements.getTypeElement(it.typeName.rawType().canonicalName)) {
```